File: python/tools/sumstruct.py
import copy

import numpy as np

def sumstruct(sin=None, sadd=None):
    if sin is None or getattr(sin, 'phot') is None:
        return sadd
    if sadd is None or getattr(sadd, 'phot') is None:
        return sin
    
    sout = sin
    fn = sin.__dict__.keys()
    for k in fn:
        saddk = getattr(sadd, k)
        if isinstance(saddk, (int, float)) or (isinstance(saddk, np.ndarray) and hasattr(saddk, '__add__') and hasattr(saddk, '__mul__')):
            # print(f"key: {k} sink: {getattr(sin, k)} saddk: {getattr(sadd, k)}")
            setattr(sout, k, getattr(sin, k) + getattr(sadd, k))
        # else:
        #     setattr(sout, k, getattr(sin, k))

    return sout

File: MATLAB/tools/sumstruct.m
function sout=sumstruct(sin, sadd)
if isempty(sin)
    sout=sadd;
    return
end
if isempty(sadd)
    sout=sin;
    return
end
fn=fieldnames(sin);
for k=1:length(fn)
    if isnumeric(sadd.(fn{k}))
        sout.(fn{k})=sin.(fn{k})+sadd.(fn{k});
    else
        sout.(fn{k})=sin.(fn{k});
    end
end
end